cmd/geoipupdate/version.go

```diff
@@ -6,7 +6,45 @@ package main
 import "runtime/debug"
 
 func init() {
-	if info, ok := debug.ReadBuildInfo(); ok && info.Main.Version != "(devel)" {
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		// Getting the build info failed, e.g., it was disabled on build.
+		return
+	}
+	if version == unknownVersion {
+		// This will set the version on go install ...
 		version = info.Main.Version
 	}
+
+	var rev, time, arch, os string
+	dirty := false
+	for _, kv := range info.Settings {
+		switch kv.Key {
+		case "vcs.revision":
+			rev = kv.Value
+		case "vcs.time":
+			time = kv.Value
+		case "vcs.modified":
+			dirty = kv.Value == "true"
+		case "GOARCH":
+			arch = kv.Value
+		case "GOOS":
+			os = kv.Value
+		}
+	}
+
+	bi := ""
+
+	if len(rev) >= 8 {
```

Comment:
What's special about number 8? :)

Reply:
This is a Git hash and we are taking the first 8 characters of it, as is common.